ingestion/src/metadata/ingestion/source/database/redshift/incremental_table_processor.py

```diff
@@ -0,0 +1,184 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Incremental Processor for Redshift
+"""
+import re
+from datetime import datetime
+from typing import Iterable, List, Tuple
+
+from sqlalchemy.engine import Connection
+
+from metadata.ingestion.source.database.redshift.models import (
+    RedshiftTable,
+    RedshiftTableChangeQueryRegex,
+    RedshiftTableMap,
+    SchemaName,
+    TableName,
+)
+from metadata.ingestion.source.database.redshift.queries import (
+    REDSHIFT_TABLE_CHANGES_QUERY,
+)
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+# ---- Regex Definitions
+TABLE_NAME_RE = r"(\w+\.){0,2}\w+"
+
+CREATE_TABLE = rf"^.*CREATE\s+(LOCAL\s+|EXTERNAL\s+)?(TEMPORARY\s+|TEMP\s+)?TABLE\s+(IF\s+NOT\s+EXISTS\s+)?(?P<table>{TABLE_NAME_RE}).*$"
```

Comment:
since we have created the `incremental_metadata.py` to work as its own class, would be good to create some tests around that. Specially since we have some rather complicated regex in Redshift, having a unit test validating a list of queries can be useful. We can then use that when we see that we are missing X or Y scenario based on user feedback

Reply:
I think it makes total sense. I'll work on adding some extra unit tests!